support-frontend/assets/pages/paper-subscription-checkout/helpers/paymentProviders.js

```diff
@@ -6,16 +6,76 @@ import {
   setupStripeCheckout,
 } from 'helpers/paymentIntegrations/stripeCheckout';
 import { type IsoCountry } from 'helpers/internationalisation/country';
-import type { PaymentAuthorisation } from 'helpers/paymentIntegrations/readerRevenueApis';
+import { type PaymentAuthorisation } from 'helpers/paymentIntegrations/readerRevenueApis';
 import {
   type PaymentResult,
+  type RegularPaymentRequest,
+  postRegularPaymentRequest, regularPaymentFieldsFromAuthorisation,
 } from 'helpers/paymentIntegrations/readerRevenueApis';
 import { type Dispatch } from 'redux';
 import { openDirectDebitPopUp } from 'components/directDebit/directDebitActions';
 import { finalPrice as paperFinalPrice } from 'helpers/productPrice/paperProductPrices';
+import { routes } from 'helpers/routes';
+import { getQueryParameter } from 'helpers/url';
+import { getOphanIds, getSupportAbTests } from 'helpers/tracking/acquisitions';
+import { Monthly } from 'helpers/billingPeriods';
+
 import { type State, setSubmissionError, setFormSubmitted, type Action, setStage } from '../paperSubscriptionCheckoutReducer';
 
-function onPaymentAuthorised(paymentAuthorisation: PaymentAuthorisation, dispatch: Dispatch<Action>) {
+function buildRegularPaymentRequest(state: State, paymentAuthorisation: PaymentAuthorisation): RegularPaymentRequest {
+  const { currencyId, countryId } = state.common.internationalisation;
+  const {
+    firstName,
+    lastName,
+    billingAddressLine1,
+    billingAddressLine2,
+    billingTownCity,
+    billingPostcode,
+    email,
+    telephone,
+  } = state.page.checkout;
+
+  const product = {
+    currency: currencyId,
+    billingPeriod: Monthly,
+    fulfilmentOptions: state.page.checkout.fulfilmentOption,
+    productOptions: state.page.checkout.productOption,
+  };
+
+  const paymentFields = regularPaymentFieldsFromAuthorisation(paymentAuthorisation);
+
+  return {
+    firstName,
+    lastName,
+    billingAddress: {
```

Comment:
this is a stylistic nit, but could we extract these as variables a bit higher up the function (like `product` is) this return looks a bit hefty to me